doc/plutus/howtos/plutus-spending-script-example.rst

```diff
@@ -0,0 +1,171 @@
+
+This is a type of Plutus script that is required to validate the
+spending of a tx output at its own script address. The tx output at the
+Plutus script address *must* be associated with a datum hash, otherwise,
+the tx output will be unspendable! The purpose of this datum hash is to
+encode the state of the contract. A demonstration of this can be seen in
+`lecture #7 <https://youtu.be/oJupInqvJUI>`__ of the Plutus Pioneers
+Program. The Plutus spending script expects a datum and a redeemer in
+order to successfully validate the spending of the tx output at its own
+script address; note that the redeemer is considered to be the user
+input. These are supplied in the transaction being submitted along with
+the Plutus script itself and specified transaction execution units.
+
+The transaction execution units are an upper bound or budget of what
+will be spent to execute the Plutus script. If you don’t specify a high
+enough value to cover script execution, your transaction will still be
+successful (provided it is a valid tx) but you will lose your
+collateral. The collateral is the transaction input(s) you specify to be
+consumed if your script fails to execute. There is a protocol parameter
+``collateralPercent`` that determines what percentage of your inputs you
+must supply as collateral.
+
+*Note that in order to use a tx input as collateral,
+it*\ **cannot**\ *reside at a script address; it must reside at a
+‘normal’ payment address and it cannot contain any multi-assets.*
+
+An example of using a Plutus spending script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below is an example that shows how to use a Plutus spending script. This
+is a step-by-step process involving:
+
+- the creation of the ``AlwaysSucceeds`` Plutus txin script
+- sending ADA to the Plutus script address
+- spending ADA at the Plutus script address
+
+In this example we will use the
+`AlwaysSucceeds <../plutus-example/plutus-example/src/Cardano/PlutusExample/AlwaysSucceeds.hs>`__
+Plutus spending script. In order to execute a Plutus spending script, we
+require the following:
+
+- Collateral tx input(s) - these are provided and are forfeited in the
+  event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx
+  output that sits at the Plutus script address. It must have a datum
+  hash, otherwise, it is unspendable.
+- The Plutus script serialized in the text envelope format.
+  ``cardano-cli`` expects Plutus scripts to be serialized in the text
+  envelope format.
+
+Creating the ``AlwaysSucceeds`` Plutus spending script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The plutus-example executable will automagically generate several Plutus
+scripts in the CLI-compatiable text envelope format.
+
+Run the following commands:
+
+.. code:: bash
+
+   cd plutus-example
+
+   cabal run exe:plutus-example
+
+This will output ``always-succeeds-txin.plutus`` in the
+``generated-plutus-scripts`` dir.
```

Comment:
Should `generated-plutus-scripts` be `.gitignored`?

Reply:
Yep :+1: